compiler/rustc_metadata/src/locator.rs

```diff
@@ -256,6 +255,7 @@ crate struct CrateLocator<'a> {
     pub triple: TargetTriple,
     pub filesearch: FileSearch<'a>,
     root: Option<&'a CratePaths>,
+    only_needs_metadata: bool,
```

Comment:
Looks like the previous placement was actually right, `only_needs_metadata` is only based on the session and not on the currently searched crate. All other fields in the "per-search configuration" apply to the specific crate that we are searching right now and can be different for other searches.

Reply:
Reverted this commit except the `PathBuf`->`Path` change.